extensions/azure-cosmos/runtime/src/main/java/io/quarkiverse/azure/cosmos/runtime/CosmosConfig.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkiverse.azure.cosmos.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "azure.cosmos", phase = ConfigPhase.RUN_TIME)
+public class CosmosConfig {
+
+    /**
+     * The flag to enable the cosmos. If set to false, the cosmos will be disabled
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean enabled;
+
+    /**
+     * The endpoint of Azure Cosmos DB. Required if quarkus.azure.cosmos.enabled is set to true
+     */
+    @ConfigItem
+    public Optional<String> endpoint;
+
+    /**
+     * The key of Azure Cosmos DB. Optional and can be empty if the Azure Identity is used to authenticate
+     */
+    @ConfigItem
+    public Optional<String> key;
```

Comment:
Can we have a sample that uses key?

Reply:
We have, see https://github.com/quarkiverse/quarkus-azure-services/pull/270/files#diff-8c48f063e81b678b3cfc7bcc9db009ab6b7d49f189c756e0afe06c8d62e93c03R41-R69.

Reply:
However, the doc only suggests using Cosmos native RBAC, see https://aka.ms/cosmos-native-rbac.